integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/deleteTable/TestDeleteTableNewDDL.scala

```diff
@@ -58,6 +65,35 @@ class TestDeleteTableNewDDL extends QueryTest with BeforeAndAfterAll {
     }.getMessage.contains("Database 'dropdb_test' not found"))
   }
 
+  test("test create database when dblocation is inconsistent") {
+    val dbName = "dropdb_test"
+    sql(s"drop database if exists $dbName cascade")
+    CarbonProperties.getInstance().addProperty(CarbonCommonConstants.STORE_LOCATION,
+      warehouse + File.separator + "carbonwarehouse")
+    val exception = intercept[InvalidOperationException] {
+      sql(s"create database $dbName")
+    }
+    assert(exception.getMessage.contains("Create database is prohibited when" +
+      " database locaton is inconsistent, please don't configure " +
+      " carbon.storelocation and spark.sql.warehouse.dir to different values"))
+    CarbonProperties.getInstance().addProperty(CarbonCommonConstants.STORE_LOCATION, warehouse)
+  }
+
+  test("test drop database when dblocation is inconsistent") {
+    val dbName = "dropdb_test"
+    sql(s"drop database if exists $dbName cascade")
+    sql(s"create database $dbName")
+    CarbonProperties.getInstance().addProperty(CarbonCommonConstants.STORE_LOCATION,
+      warehouse + File.separator + "carbonwarehouse")
+    val exception = intercept[InvalidOperationException] {
```

Comment:
after create DB, if store location is changed, create table will create empty directory in store location path. On drop database, since you are throwing exception, On next drop db (when store location and spark.warehouse.dir is same) the directory present in store location will be not be deleted. I feel we can only restrict create database, and not throw exception with drop database. @kunal642 what is your thought about this ?

Reply:
I will raise a pr to avoid mk empty dir under table creation.

Reply:
please raise JIRA to track it